### PR TITLE
Enabling symlink creation on Windows

### PR DESCRIPTION
**Pull request by brawner**
This adds symlink creation for Windows for 'latest {Verb}'. Currently os.symlink only works when run with administrator privileges and it will throw an OSError if symlink privileges are not available.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Please address the linter warning and update the corresponding test to not skip Windows.

**Comment by brawner** [comment]
Is it ok to assume the test will be run with admin privileges on Windows?

**Comment by dirk-thomas** [comment]
I don't know what AppVeyor uses.

File: colcon_core/location.py [hunk 1]
```diff
@@ -121,11 +120,13 @@ def create_log_path(verb_name):
     A `COLCON_IGNORE` marker file is being placed in the parent directory of
     the logging directory to avoid it being crawled for packages.
 
-    On non-Windows platforms two symlinks are created as siblings of the log
-    path:
+    Two symlinks are created as siblings of the log path:
     * `latest_<verb_name>` linking to the log path
     * `latest` linking to `latest_<verb_name>`
 
+    On Windows platforms, Administrator privilges are required to create
```
Comment (dirk-thomas) [inline colcon_core/location.py]: Spelling: "privilges"

Comment (brawner) [reply colcon_core/location.py]:
```suggestion
    On Windows platforms, Administrator privileges are required to create
```

Comment (brawner) [reply colcon_core/location.py]: Done

**Comment by davetcoleman** [comment]
Great contribution!

**Comment by brawner** [comment]
Thanks Dave!

**Comment by dirk-thomas** [comment]
Can you please update the test to also mock the `symlink` function and raise the expected exception to cover these new lines.

**Comment by dirk-thomas** [comment]
@brawner Friendly ping.

**Comment by brawner** [comment]
@dirk-thomas In order to raise the exception, the user must be demoted to standard privileges and I cannot find a way to do this on Windows in the middle of a Python script. Changing permissions in the directory to read-only will raise a different exception (PermissionsError).

Are there any other alternatives I'm not thinking about?

**Comment by dirk-thomas** [comment]
> Are there any other alternatives I'm not thinking about?

You can mock the Python function to raise the expected exception.

**Comment by dirk-thomas** [comment]
Thank you for the patch.

**Pull request merged**